compiler/rustc_errors/src/json.rs

```diff
@@ -135,6 +141,26 @@ impl Emitter for JsonEmitter {
         }
     }
 
+    fn emit_timing_section(&mut self, section: TimingSection, kind: TimingSectionKind) {
+        let kind = match kind {
+            TimingSectionKind::Start => "start",
+            TimingSectionKind::End => "end",
+        };
```

Comment:
This could be a method on `TimingSectionKind`.

Reply:
Hmm, but the `TimingSectionKind` is a shared data structure for all emitters, and this form of output is specific for JSON. It should IMO live here, in theory we could emit the timings also in other formats. It also avoid having to go to a different file to see how the output format looks like.

I'm fine with moving it to a function in the `json` module, but I don't think it belongs as a method on the section kind.